hello_test.cc

```diff
@@ -1,10 +1,155 @@
-}
-
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+
+#include "element.h"
+#include "board.h"
+#include "movemap.h"
+
+TEST(Sphere, Constructors) {
+  Sphere sphere;
+
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere.at(element), 0);
+  }
+
+  Sphere sphere2({Element::FIRE});
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere2.at(element), element == Element::FIRE);
+  }
+
+  Sphere sphere3({{Element::WATER, 1}, {Element::EARTH, 2}});
+  for (auto element : ELEMENTLIST) {
+    if (element == Element::WATER) {
+      EXPECT_EQ(sphere3.at(element), 1);
+    }
+    else if (element == Element::EARTH)
+    {
+      EXPECT_EQ(sphere3.at(element), 2);
+    }
+    else {
+      EXPECT_EQ(sphere3.at(element), 0);
+    }
+  }
+
+}
+
+TEST(Sphere, CountTest) {
+  Sphere sphere;
+
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere.count(element), 0);
+  }
+
+  Sphere sphere2({Element::FIRE});
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(sphere2.count(element), element == Element::FIRE);
+  }
+
+  Sphere sphere3({{Element::WATER, 1}, {Element::EARTH, 2}});
+  for (auto element : ELEMENTLIST) {
+    if (element == Element::WATER) {
+      EXPECT_EQ(sphere3.count(element), 1);
+    }
+    else if (element == Element::EARTH)
+    {
+      EXPECT_EQ(sphere3.count(element), 2);
+    }
+    else {
+      EXPECT_EQ(sphere3.count(element), 0);
+    }
+  }
+  EXPECT_EQ(sphere3.count(), 3);
+}
+
+
+
+TEST(Board, BoardInitGemCounts) {
+  Board board;
+  
+  for (auto element : ELEMENTLIST) {
+    EXPECT_EQ(board.count(element, element), 1);
+    EXPECT_EQ(board.count(element, ~element), 0);
+  }
+
+}
+
+TEST(Board, MoveMapTest) {
+  std::vector<std::tuple<Element, Element, Element>> moves = {{Element::FIRE, Element::FIRE, Element::WATER}, {Element::EARTH, Element::EARTH, Element::LIGHTNING}, {Element::EARTH, Element::EARTH, Element::LIGHTNING}};
+  
+  MoveMap moveMap(moves);
+  EXPECT_EQ(moveMap.inputCount(), 3);
+  EXPECT_EQ(moveMap.inputCount(Element::EARTH, Element::EARTH), 2);
+  EXPECT_EQ(moveMap.outputCount(), 3);
+  EXPECT_EQ(moveMap.outputCount(Element::LIGHTNING, Element::EARTH), 2);
+}
+
+TEST(Board, SetSphereTest) {
+  Board board;
+  Sphere expectedSphere({Element::FIRE});
+
+  EXPECT_TRUE(board.getSphere(Element::FIRE).equal(expectedSphere));
+
+  Sphere sphere({Element::FIRE, Element::FIRE, Element::LIGHTNING});
+  board.setSphere(Element::FIRE, sphere);
+  expectedSphere = Sphere({Element::FIRE, Element::FIRE, Element::LIGHTNING});
+  
+  EXPECT_TRUE(board.getSphere(Element::FIRE).equal(expectedSphere));
+
+}
+
+
+TEST(Board, CountTest) {
+  Board board;
+  
+  Sphere sphere({Element::FIRE, Element::FIRE, Element::LIGHTNING});
+  board.setSphere(Element::FIRE, sphere);
+  
+  EXPECT_EQ(board.count(Element::FIRE, Element::FIRE), 2);
+  EXPECT_EQ(board.count(Element::FIRE, Element::LIGHTNING), 1);
+  EXPECT_EQ(board.count(Element::FIRE, Element::ALL), 3);
+  EXPECT_EQ(board.count(), 7);
+}

```

Comment:
This looks like 2 tests to me. 1 for counting specific elements, one for all elements.